installer/bundle/bundle_skel_SunOS.sh

```diff
@@ -0,0 +1,432 @@
+
+echo "mail=" > scx-admin-upgrade
+echo "instance=overwrite" >> scx-admin-upgrade
+echo "partial=nocheck" >> scx-admin-upgrade
+echo "idepend=quit" >> scx-admin-upgrade
+echo "rdepend=nocheck" >> scx-admin-upgrade
+echo "conflict=nocheck" >> scx-admin-upgrade
+echo "action=nocheck" >> scx-admin-upgrade
+echo "basedir=default" >> scx-admin-upgrade
+
+# Do we need to remove the package?
+if [ "$installMode" = "R" -o "$installMode" = "P" ]
+then
+    if [ -f /opt/microsoft/scx/bin/uninstall ]; then
+        /opt/microsoft/scx/bin/uninstall $installMode
+    else
+        pkg_rm scx
+        pkg_rm omi
+    fi
+
+    if [ "$installMode" = "P" ]
+    then
+        echo "Purging all files in cross-platform agent ..."
+        rm -rf /etc/opt/microsoft/scx /opt/microsoft/scx /var/opt/microsoft/scx
+        rmdir /etc/opt/microsoft /opt/microsoft /var/opt/microsoft 1>/dev/null 2>/dev/null
+
+        # If OMI is not installed, purge its directories as well.
+        check_if_pkg_is_installed omi
+        if [ $? -ne 0 ]; then
+            rm -rf /etc/opt/omi /opt/omi /var/opt/omi
+        fi
+    fi
+fi
+
+if [ -n "${shouldexit}" ]
+then
+    # when extracting script/tarball don't also install
+    cleanup_and_exit 0
+fi
+
+#
+# Do stuff before extracting the binary here, for example test [ `id -u` -eq 0 ],
+# validate space, platform, uninstall a previous version, backup config data, etc...
+#
+
+#
+# Extract the binary here.
+#
+
+echo "Extracting..."
+tail +${SCRIPT_LEN_PLUS_ONE} "${SCRIPT}" | zcat - | tar xf -
+STATUS=$?
+if [ ${STATUS} -ne 0 ]
+then
+    echo "Failed: could not extract the install bundle."
+    cleanup_and_exit ${STATUS}
+fi
+
+#
+# Do stuff after extracting the binary here, such as actually installing the package.
+#
+
+EXIT_STATUS=0
+SCX_EXIT_STATUS=0
+OMI_EXIT_STATUS=0
+
+case "$installMode" in
+    E)
+        # Files are extracted, so just exit
+        cleanup_and_exit 0 "SAVE"
+        ;;
+
+    I)
+        echo "Installing cross-platform agent ..."
+        check_if_pkg_is_installed scx
+        if [ $? -eq 0 ]; then
+            echo "ERROR: SCX package is already installed"
+            cleanup_and_exit 2
+        fi
+
+        check_if_pkg_is_installed omi
+        if [ $? -eq 0 ]; then
+            pkg_upd $OMI_PKG omi
+            # It is acceptable that this fails due to the new omi being
+            # the same version (or less) than the one currently installed.
+            OMI_EXIT_STATUS=0
+        else
+            pkg_add $OMI_PKG omi
+            OMI_EXIT_STATUS=$?
+        fi
+
+        pkg_add $OM_PKG scx
+        SCX_EXIT_STATUS=$?
+
+        ;;
+
+    U)
+        echo "Updating cross-platform agent ..."
+        check_if_pkg_is_installed omi
+        if [ $? -eq 0 ]; then
+            pkg_upd $OMI_PKG omi
+            # It is acceptable that this fails due to the new omi being
+            # the same version (or less) than the one currently installed.
+            OMI_EXIT_STATUS=0
```

Comment:
Ditto with above, I wouldn't think this makes sense here given that pkg_upd does version comparison. Depending on how robust that is, you may not even need to check if OMI is installed (although I'd guess it's not that robust). But definitely, if that code all works "more or less properly", I think we should be picking up the exit status code properly and using it to determine our exit status.